File: src/main/kotlin/exercises/day1/part1.kt
package exercises.day1

import common.Reader

class Comparer {
  fun compare(first: Int, second: Int): Trend
      = when {
    second > first -> Trend.INCREASE
    second < first -> Trend.DECREASE
    else -> Trend.NONE
  }
}

enum class Trend {
  INCREASE, DECREASE, NONE
}

fun main() {
  val reader = Reader()
  val comparer = Comparer()

  val lines = reader.readFile("/Users/serafinodangelillo/Workspace/AdventOfCode2021/src/main/resources/day1/input")

  var increments = 0
  for (n in 0..lines.size-2) {
    val currentLine = lines[n].toInt()
    val nextLine = lines[n+1].toInt()

    if(comparer.compare(currentLine, nextLine) == Trend.INCREASE)
      increments++
  }

  println(increments)
}

File: src/main/haskell/exercises/day1/Part1.hs
module Exercises.Day1.Part1 (
  executePart1
) where


convertToInt :: String -> Int
convertToInt a = read a :: Int

getIntLines :: String -> [Int]
getIntLines = map convertToInt . lines


executePart1 :: IO ()
executePart1 = do raw <- readFileContent
                  mapM_ print (getIntLines raw)

readFileContent :: IO String
readFileContent = readFile "src/main/resources/day1/input"
